tests/positive/general/preemption.go

```diff
@@ -74,14 +74,26 @@ func makePreemptTest(components string) types.Test {
 	}
 
 	var systemFiles []types.File
+	var systemFile types.File
 	for _, component := range []string{"b", "u"} {
 		if enabled(component) {
-			systemFiles = append(systemFiles, types.File{
-				Node: types.Node{
-					Name: longnames[component] + ".ign",
-				},
-				Contents: makeConfig(component),
-			})
+			if component == "b" {
+				systemFile = types.File{
```

Comment:
This is still more redundant than necessary.  You can just conditionalize a variable holding the field value for `Directory`.